src/widget.cpp

```diff
@@ -254,9 +255,39 @@ void widget::load( const JsonObject &jo, const std::string & )
     optional( jo, was_loaded, "widgets", _widgets, string_id_reader<::widget> {} );
 }
 
+static int set_widget_label_width( const widget_id &id )
```

Comment:
Maybe I'm just dense, but is there a reason this is a separate static function, instead of being inside the `widget` class?

Reply:
It's just a convenience function for stuff in `widget::finalize()` and is not intended to be shared outside of this one file. I guess the name might be a little confusing, didn't really know what to call it other that `finalize_helper_for_labels` :P

Reply:
Could it be a static class function maybe, like `finalize` itself is? I'm just thinking static (non-class) functions are harder to unit test, and it's tightly coupled to the widget class, so it may as well be part of the class. Or, just fold it into `widget::finalize` itself, since there's naught else in there (and it'd avoid doubly iterating over `widget::get_all()`).

Reply:
> Could it be a static class function maybe, like `finalize` itself is?

Sure, I can do that.

> Or, just fold it into `widget::finalize` itself, since there's naught else in there (and it'd avoid doubly iterating over `widget::get_all()`)

Not possible unfortunately, it has to be its own function since it calls itself recursively to handle nested layouts (kinda similar to `set_height_for_widget`).

Also note that, because of how widgets are self-modifying, it needs to obtain the original object from the widget factory to persist the changes. That was a real head-scratcher for me :P

Reply:
Oh, I missed the recursive part. And somehow I'm not surprised about the factory part, that's a tricky beast to work with sometimes.